### PR TITLE
Google Seeds

### PR DESCRIPTION
**Pull request by slemrmartin**
Source Type and ApplicationType seeds for Google Cloud Engine

* [x] **depends on** https://github.com/RedHatInsights/platform-mq/pull/114

---

[RHCLOUD-11249](https://issues.redhat.com/browse/RHCLOUD-11249)

### CODE REVIEW

**Comment by slemrmartin** [comment]
cc @dccurtis

**Comment by slemrmartin** [comment]
@dccurtis availability check url will be the same like for aws? `/api/cost-management/v1/source-status/`

**Comment by dccurtis** [comment]
> @dccurtis availability check url will be the same like for aws? `/api/cost-management/v1/source-status/`

Yes, no difference there

File: db/seeds/source_types.yml [hunk 1]
```diff
@@ -123,6 +123,32 @@ ansible-tower:
...
           :when: endpoint.verify_ssl
           :is: true
+google:
+  :product_name: Google Cloud Engine
+  :vendor: Google
+  :icon_url: "/apps/chrome/assets/images/partners-icons/google-cloud-engine.svg" # TODO
+  :schema:
+    :authentication:
+    - :type: username_password
+      :name: Project ID and Service Account JSON
```
Comment (dccurtis) [inline db/seeds/source_types.yml:133]: is the Service Account JSON going to be used to seed the service account email for the wizard?

File: db/seeds/source_types.yml [hunk 2]
```diff
@@ -123,6 +123,32 @@ ansible-tower:
...
+      :fields:
+      - :component: text-field
+        :name: authentication.authtype
+        :hideField: true
+        :initializeOnMount: true
+        :initialValue: username_password
```
Comment (dccurtis) [inline db/seeds/source_types.yml:139]: Just want to make sure I understand: `project_id` will go in the `Authentications` `authtype` field?

File: db/seeds/source_types.yml [hunk 3]
```diff
@@ -123,6 +123,32 @@ ansible-tower:
...
+        :label: Project ID
+      - :component: textarea
+        :name: authentication.password
+        :label: Service Account JSON
+    :endpoint:
```
Comment (dccurtis) [inline db/seeds/source_types.yml:144]: Would we be able to leverage the new `Application` `extra` field to store the `dataset`?

Something like this
```
      - :name: application.extra.dataset
        :stepKey: cost-management
        :component: text-field
        :label: Dataset name
        :isRequired: true
```

File: db/seeds/source_types.yml (hunk 2 above)
Comment (slemrmartin) [reply db/seeds/source_types.yml:139]: no, that's mistake :)

File: db/seeds/source_types.yml (hunk 1 above)
Comment (slemrmartin) [reply db/seeds/source_types.yml:133]: I don't know, but in the credentials provided by QE there is a `client_email` key-value in the JSON

Comment (slemrmartin) [reply db/seeds/source_types.yml:133]: Anyway, I started from the ManageIQ's create form: https://files.gitter.im/5aa15e4ad73408ce4f907617/dsFs/Screenshot-2021-01-15-at-10.44.16.png

Comment (dccurtis) [reply db/seeds/source_types.yml:133]: Are there other use cases for GCP that you are working on besides Cost Management?

Comment (slemrmartin) [reply db/seeds/source_types.yml:133]: There is and oportunity for topological-inventory collector, but we don't have any milestone for that as far as i know

File: db/seeds/source_types.yml (hunk 3 above)
Comment (rvsia) [reply db/seeds/source_types.yml:146]: Yes, I agree! 👍  We will do it anyway in the future and doing it now will prevent any painful migration.

(You should also provide a validate in that object:

```
        :validate:
        - :type: required
```

`isRequired` is just a boolean to show the required marker.)

File: db/seeds/source_types.yml [hunk 4]
```diff
@@ -123,6 +123,34 @@ ansible-tower:
         :condition:
           :when: endpoint.verify_ssl
           :is: true
+google:
+  :product_name: Google Cloud Engine
+  :vendor: Google
+  :icon_url: "/apps/chrome/assets/images/partners-icons/google-cloud-engine.svg" # TODO
```
Comment (rvsia) [inline db/seeds/source_types.yml:129]: I have already uploaded the logo: '/apps/frontend-assets/partners-icons/google-cloud.svg'

File: db/seeds/source_types.yml [hunk 5]
```diff
@@ -123,6 +123,34 @@ ansible-tower:
         :condition:
           :when: endpoint.verify_ssl
           :is: true
+google:
+  :product_name: Google Cloud Engine
```
Comment (rvsia) [inline db/seeds/source_types.yml:127]: I think we should discuss with UX how to call it, this value is then shown in the ui in multiple places. @beccaglass

File: db/seeds/source_types.yml [hunk 6]
```diff
@@ -123,6 +123,34 @@ ansible-tower:
...
+  :vendor: Google
+  :icon_url: "/apps/chrome/assets/images/partners-icons/google-cloud-engine.svg" # TODO
+  :schema:
+    :authentication:
+    - :type: username_password
```
Comment (rvsia) [inline db/seeds/source_types.yml:132]: I think we should describe it more accurately as we don't actually provide any username or password - `project_id` maybe ?.

File: db/seeds/source_types.yml (hunk 5 above)
Comment (beccaglass) [reply db/seeds/source_types.yml:127]: My preference is 'Google Cloud' -- seems to be the most scalable

File: db/seeds/source_types.yml (hunk 3 above)
Comment (slemrmartin) [reply db/seeds/source_types.yml:153]: added

File: db/seeds/source_types.yml (hunk 6 above)
Comment (slemrmartin) [reply db/seeds/source_types.yml]: changed to `project_id_service_account_json`

**Review by rvsia — APPROVED** [review]
:+1:

**Comment by dccurtis** [comment]
Thanks for adding the extra field for dataset!  I'll update on my side to query for `project_id_service_account_json` to get the `project_id` from the username field 👍

**Review by lindgrenj6 — APPROVED** [review]

**Pull request merged**